src/mds/MDSDaemon.cc

```diff
@@ -472,13 +472,14 @@ int MDSDaemon::init(MDSMap::DaemonState wanted_state)
       dout(0) << "Specified oneshot replay mode but not an MDS!" << dendl;
       suicide();
     }
+    wanted_state = MDSMap::STATE_BOOT;
     standby_type = wanted_state;
```

Comment:
I'll need to look at this in more detail for the fiddly bits, but do we need to set standby_type before changing wanted_state here? Otherwise I'm not sure how standby_type can be STATE_STANDBY_REPLAY below.